platforms/ios/lib/WysiwygComposer/Sources/HTMLParser/HTMLParser.swift

```diff
@@ -107,15 +104,29 @@ public final class HTMLParser {
                 mutableAttributedString.addAttributes([.foregroundColor: style.linkColor], range: range)
             }
         }
-
+        
         mutableAttributedString.applyBackgroundStyles(style: style)
         mutableAttributedString.applyInlineCodeBackgroundStyle(codeBackgroundColor: style.codeBackgroundColor)
-
+        
         // FIXME: This solution might not fit for everything.
         mutableAttributedString.addAttribute(.paragraphStyle,
                                              value: NSParagraphStyle.default,
                                              range: .init(location: 0, length: mutableAttributedString.length))
-
+        
+        removeTrailingNewlineIfNeeded(from: mutableAttributedString, given: html)
         return mutableAttributedString
     }
+    
+    private static func removeTrailingNewlineIfNeeded(from mutableAttributedString: NSMutableAttributedString, given html: String) {
+        // DTCoreText always adds a \n at the end of the document, which we need to remove
+        // however it does not add it if </code> </a> are the last nodes.
+        // Also we don't want to remove if a blockquote or a codeblock have more than one paragraph
+        if mutableAttributedString.string.last == "\n",
+           !html.hasSuffix("</code>"),
+           !html.hasSuffix("</a>"),
+           !html.hasSuffix("</p><p></p></blockquote>"),
+           !html.hasSuffix("\n</pre>") {
```

Comment:
That's... oddly specific 😅 . What happens if the last paragraph in either the blockquote or codeblock is not empty, or if the previous item in a blockquote is a list?

Reply:
if the paragraph is not empty nothing will happen, this is just to prevent iOS from essentially not entering a newline when is actually needed.
Without this check is impossible to enter a newline after writing some stuff in a blockquote or a codeblock, because we are removing that same newline which is actually a needed one.

For the list case you're right let me check it!

Reply:
Yeah the list case had some issues! added it, also improved the comment to explain better the reasoning for the case in which we do not want to remove that \n

Reply:
```
// DTCoreText always adds a \n at the end of the document, which we need to remove
// however it does not add it if </code> </a> are the last nodes.
 // Also we don't want to remove it if blockquote and codeblock contain that newline
// and are not empty, because DTCoreText does not add a newline if these blocks
// contain one at the end.
```